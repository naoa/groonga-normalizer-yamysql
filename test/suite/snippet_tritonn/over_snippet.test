register normalizers/yamysql
register functions/snippet_tritonn

table_create remove_phrases TABLE_PAT_KEY ShortText
load --table remove_phrases
[
{"_key": "<remove_symbol>"}
]

normalize NormalizerYaMySQL "実践!!C++" WITH_TYPES|WITH_CHECKS

table_create Entries TABLE_NO_KEY
column_create Entries content COLUMN_SCALAR ShortText

table_create Tokens TABLE_PAT_KEY ShortText --default_tokenizer TokenBigram --normalizer NormalizerYaMySQL
column_create Tokens entries_content COLUMN_INDEX|WITH_POSITION Entries content

load --table Entries
[
{"content": "実践!!C++入門"}
]

select Entries \
  --output_columns 'snippet_tritonn(content, 200, 2, \
  "NormalizerYaMySQL", \
  0, 0, "...", "...", \
  "C",  "<span class=\\"w1\\">", "</span>"), content' \
  --command_version 2 \
  --match_columns 'content' \
  --query "C"
