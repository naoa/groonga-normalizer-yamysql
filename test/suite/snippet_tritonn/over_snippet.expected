register normalizers/yamysql
[[0,0.0,0.0],true]
register functions/snippet_tritonn
[[0,0.0,0.0],true]
table_create remove_phrases TABLE_PAT_KEY ShortText
[[0,0.0,0.0],true]
load --table remove_phrases
[
{"_key": "<remove_symbol>"}
]
[[0,0.0,0.0],1]
normalize NormalizerYaMySQL "実践!!C++" WITH_TYPES|WITH_CHECKS
[
  [
    0,
    0.0,
    0.0
  ],
  {
    "normalized": "実践 c ",
    "types": [
      "kanji",
      "kanji",
      "null",
      "alpha",
      "null"
    ],
    "checks": [
      3,
      0,
      0,
      3,
      0,
      0,
      2,
      1,
      2
    ]
  }
]
table_create Entries TABLE_NO_KEY
[[0,0.0,0.0],true]
column_create Entries content COLUMN_SCALAR ShortText
[[0,0.0,0.0],true]
table_create Tokens TABLE_PAT_KEY ShortText --default_tokenizer TokenBigram --normalizer NormalizerYaMySQL
[[0,0.0,0.0],true]
column_create Tokens entries_content COLUMN_INDEX|WITH_POSITION Entries content
[[0,0.0,0.0],true]
load --table Entries
[
{"content": "実践!!C++入門"}
]
[[0,0.0,0.0],1]
select Entries   --output_columns 'snippet_tritonn(content, 200, 2,   "NormalizerYaMySQL",   0, 0, "...", "...",   "C",  "<span class=\\"w1\\">", "</span>"), content'   --command_version 2   --match_columns 'content'   --query "C"
[
  [
    0,
    0.0,
    0.0
  ],
  [
    [
      [
        1
      ],
      [
        [
          "snippet_tritonn",
          "Bool"
        ],
        [
          "content",
          "ShortText"
        ]
      ],
      [
        [
          "...実践!!<span class=\"w1\">C</span>++入門..."
        ],
        "実践!!C++入門"
      ]
    ]
  ]
]
